File: Array/longest-subarray-of-1s-after-deleting-one-element/submission-0.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_num = 0
        for index, val in enumerate(nums):
            if val == 0 and (index > 0) and (index < (len(nums)-1)) and nums[index-1] == 1 and nums[index+1] == 1:
                temp = list(nums)
                temp.pop(index)
                max_num = max(self.longest1s(temp), max_num)
            elif index == 0 or  index == (len(nums)-1):
                temp = list(nums)
                temp.pop(index)
                max_num = max(self.longest1s(temp), max_num)

        for index, val in enumerate(nums):
            if val == 1:
                temp = list(nums)
                temp.pop(index)
                max_num = max(self.longest1s(temp), max_num)
                break
                
        return max_num

    def longest1s(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_num = 0
        counter = 0
        for val in nums:
            if val == 1:
                counter+= 1
            else:
                if counter > max_num:
                    max_num = counter
                counter = 0
        
        if counter > max_num:
            max_num = counter
        
        return max_num

        

File: Array/smallest-stable-index-ii/submission-0.cpp
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        auto size_nums = static_cast<int>(nums.size());
        auto preMin = std::vector<int>(size_nums);
        preMin[size_nums-1] = nums[size_nums-1];
        for (auto i = size_nums - 2; i >= 0; i--) {
            if (preMin[i+1] > nums[i]) {
                preMin[i] = nums[i];
            } else {
                preMin[i] = preMin[i+1];
            }
        }

        auto max = nums[0];
        for (auto i = 0; i < size_nums; i++) {
            if (max < nums[i]) {
                max = nums[i];
            }

            if (max - preMin[i] <= k) {
                return i;
            }
        }

        return -1;
    }
};